api/server/v1alpha1/device_types.go

```diff
@@ -128,9 +131,36 @@ func (c *Device) SetReclaimPolicy(p runtimev1alpha1.ReclaimPolicy) {
 
 // +kubebuilder:object:root=true
 
-// DeviceList contains a list of Device
+// DeviceList contains a list of Devices
 type DeviceList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Device `json:"items"`
 }
+
+// DeviceClassSpecTemplate ...
+type DeviceClassSpecTemplate struct {
+	runtimev1alpha1.NonPortableClassSpecTemplate `json:",inline"`
+	DeviceSpec                                   `json:",inline"`
```

Comment:
Because `DeviceSpec` has `runtimev1alpha1.ResourceSpec` embedded in it and the `DeviceClass` is not a managed resource (it is a non-portable class), we usually separate out the fields of the provider-specific spec fields into a different struct called something like `DeviceParameters` and embed that in the `DeviceSpec` and `DeviceClassSpecTemplate`.

Ref: https://github.com/crossplaneio/stack-gcp/blob/32c2999e25af12712aa811658c4526d084838944/apis/cache/v1alpha2/cloudmemorystore_instance_types.go#L109

Ref: https://github.com/crossplaneio/stack-gcp/blob/32c2999e25af12712aa811658c4526d084838944/apis/cache/v1alpha2/cloudmemorystore_instance_types.go#L172

Ref: https://github.com/crossplaneio/stack-gcp/blob/32c2999e25af12712aa811658c4526d084838944/apis/cache/v1alpha2/cloudmemorystore_instance_types.go#L46

Reply:
I've added requested changes essentially I encapsulated `DeviceParameters` and used it in `DeviceClassTemplate` and `DeviceSpec`.